src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Unix/OpenSslX509Encoder.cs

```diff
@@ -18,7 +18,7 @@ public AsymmetricAlgorithm DecodePublicKey(Oid oid, byte[] encodedKeyValue, byte
             switch (oid.Value)
             {
                 case Oids.RsaRsa:
-                    return BuildRsaPublicKey(encodedKeyValue);
+                    return DecodePublicKey(oid, encodedKeyValue, encodedParameters);
```

Comment:
Why do we need the switch in both places?  Since `BuildRsaPublicKey` is still around, why not just go direct from here?